### PR TITLE
R cmd notes

### LINKED ISSUE #671: [feature:] Warnings and Notes from R CMD check
### Describe the situation that led to the request and a solution

I'm wondering if the number of Warnings and Notes from R CMD check could be reduced. This could make it easier for developers/contributors to see when new code introduces relevant warnings or notes.

### Alternative solutions

_No response_

### Statistical validity

_No response_

### Additional context

As of 2 Oct 2024, these are the results of R CMD check on the main branch:

* checking C++ specification ... NOTE
```
  Specified C++11: please drop specification unless essential
```

* checking installed package size ... NOTE
```
  installed size is 53.1Mb
```

* checking dependencies in R code ... NOTE
```
Namespace in Imports field not imported from: ‘scales’
```
* checking for missing documentation entries ... WARNING
```
Undocumented code objects:
  ‘AgeComp’ ‘BevertonHoltRecruitment’ ‘CreateTMBModel’
  ‘DoubleLogisticSelectivity’ ‘EWAAgrowth’ ‘Fleet’ ‘Index’
  ‘LogisticMaturity’ ‘LogisticSelectivity’ ‘Parameter’ ‘Population’
  ‘TMBDlnormDistribution’ ‘TMBDmultinomDistribution’
  ‘TMBDnormDistribution’ ‘clear’ ‘clear_logs’ ‘get_fixed’ ‘get_random’
All user-level objects in a package should have documentation entries.
```
* checking compilation flags in Makevars ... WARNING
```
Non-portable flags in variable 'PKG_CXXFLAGS':
  -w
```
* checking compiled code ... WARNING
```
File ‘FIMS/libs/FIMS.so’:
  Found ‘_ZSt4cout’, possibly from ‘std::cout’ (C++)
    Object: ‘FIMS.o’
  Found ‘exit’, possibly from ‘exit’ (C)
    Object: ‘FIMS.o’
Compiled code should not call entry points which might terminate R nor write to
stdout/stderr instead of to the console, nor use Fortran I/O nor system RNGs nor
[v]sprintf.
See ‘Writing portable packages’ in the ‘Writing R Extensions’ manual.
```
Status: 3 WARNINGs, 3 NOTEs

### PR DESCRIPTION
**Pull request by kellijohnson-NOAA**
# What is the feature?
Changes to reduce number of warnings and notes from R CMD check from 3 and 4 to 1 and 2, respectively

# How have you implemented the solution?
* use `x, y, ...` instead of just `x` even though we only need x arguments to the method of `plot()` for FIMSFrame to match plot.
* Moves `setMethods()` out of operators.R to zzz.R to eliminate warnings that are found in R CMD check
* Utilize @rdname to combine documentation
* Format calls to `methods::setMethods()`
* adds i = 0 for rccp stuff to initialize
* changes makevars to remove warnings from R on windows when compiling

# Does the PR impact any other area of the project, maybe another repo?
Will close PR #672 because changing the call to plot is more robust than exporting scales.

Big thanks to @arni-magnusson for mentioning how important it is to keep these notes and warnings to a minimum so new contributors are not overloaded with warnings when trying to compile and can see if their changes are making a difference.

@Andrea-Havron-NOAA can you give this a quick glance?

Close #671

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -79,6 +79,7 @@ Suggests:
     rmarkdown,
     snowfall,
     testthat (>= 3.0.0),
+    tidyr,
     tidyverse,
```
Comment (Andrea-Havron-NOAA) [inline DESCRIPTION:83]: tidyverse maintainers have asked R package developers to _not_ add tidyverse to Imports, Depends, or Suggests in an R package (see [here](https://www.tidyverse.org/blog/2018/06/tidyverse-not-for-packages/)). The preferred approach is to list specific packages used within the tidyverse instead.

File: src/Makevars [hunk 2]
```diff
@@ -1,4 +1,4 @@
 CXX_STD = CXX17
-PKG_CXXFLAGS = -std=c++17  -DTMB_MODEL  -DTMB_EIGEN_DISABLE_WARNINGS -w
+PKG_CXXFLAGS = -DTMB_MODEL  -DTMB_EIGEN_DISABLE_WARNINGS
 CXX17STD = -std=c++17 -w
-
+USE_CXX17 = "yes"
```
Comment (Andrea-Havron-NOAA) [inline src/Makevars:4]: does this line need to be added to Makevars.win?

**Review by Andrea-Havron-NOAA — APPROVED** [review]

File: src/Makevars (hunk 2 above)
Comment (kellijohnson-NOAA) [reply src/Makevars:4]: The R CMD check for windows, linux, and mac currently states
```
* checking C++ specification ... OK
  Not all R platforms support C++17
```
with the current configuration so I believe the files are fine as is.

File: DESCRIPTION (hunk 1 above)
Comment (kellijohnson-NOAA) [reply DESCRIPTION:83]: Thanks @Andrea-Havron-NOAA for noticing this, I will remove tidyverse.

Comment (kellijohnson-NOAA) [reply DESCRIPTION]: Thanks again @Andrea-Havron-NOAA I also noticed that we needed {purrr} so I added that. And, I added a discussion about {processR} which is used in the tests but that does not need to be decided today so I am going to go ahead and merge in this PR.

**Comment by kellijohnson-NOAA** [comment]
Thank you @Andrea-Havron-NOAA for the quick review, super helpful as I am trying to get each new branch merged in daily for these small trivial changes so people can branch off of dev daily without coming into merge conflicts at the end of the day.

**Pull request merged**